engine/render/src/render/render_script.cpp

```diff
@@ -582,6 +585,24 @@ namespace dmRender
         {0, 0}
     };
 
+    static inline dmGraphics::BufferType CheckBufferType(lua_State* L, int index)
+    {
+        dmGraphics::BufferType buffer_type = (dmGraphics::BufferType) luaL_checkinteger(L, index);
+
+        if (!(buffer_type == dmGraphics::BUFFER_TYPE_COLOR0_BIT ||
+              buffer_type == dmGraphics::BUFFER_TYPE_COLOR1_BIT ||
+              buffer_type == dmGraphics::BUFFER_TYPE_COLOR2_BIT ||
+              buffer_type == dmGraphics::BUFFER_TYPE_COLOR3_BIT ||
+              buffer_type == dmGraphics::BUFFER_TYPE_DEPTH_BIT  ||
+              buffer_type == dmGraphics::BUFFER_TYPE_STENCIL_BIT))
+        {
+            luaL_error(L, "Unknown buffer type supplied (%d).", (int) buffer_type);
```

Comment:
The idiom is to "return luaL_error()".
It doesn't actually return, since it longjumps, but that's what they suggest in Lua docs.